addons/google_calendar/models/google_sync.py

```diff
@@ -166,20 +165,67 @@ def _sync_google2odoo(self, google_events: GoogleEvent, default_reminders=()):
 
         return synced_records
 
+    def _google_error_handling(self, http_error):
+        # We only handle the most problematic errors of sync events.
+        if http_error.response.status_code in (403, 400):
+            response = http_error.response.json()
+            message = response['error'].get('message')
+            if self._name == 'calendar.event':
+                start = self.start.strftime('%Y-%m-%d at %H:%M')
+                event_ids = self.id
+                name = self.name
+                error_log = "Error while syncing event: "
+                event = self
+            else:
+                # calendar recurrence is triggering the error
+                start = self.base_event_id.start.strftime('%Y-%m-%d at %H:%M')
+                event_ids = _("%(id)s and %(length)s following", id=self.base_event_id.id, length=len(self.calendar_event_ids.ids))
+                name = self.base_event_id.name
+                # prevent to sync other events
+                self.calendar_event_ids.need_sync = False
+                error_log = "Error while syncing recurrence: "
+                event = self.base_event_id
+
+            # We don't have right access on the event or the request paramaters were bad.
+            if http_error.response.status_code == 400 and "invalid attendee email" in message.lower():
+                reason = "invalid attendee email. Please, verify the partners emails."
+            elif http_error.response.status_code == 403 and "forbiddenForNonOrganizer" in http_error.response.text:
+                reason = "you don't have write access on this event. Please, check if you have right access on Google for this particular event"
+            else:
+                reason = response['error'].get('message')
```

Comment:
Any reference to those documented status codes you could mention here? that's a lot of super-specific hardcoded values.

Reply:
also, the reasons are hardcoded in english and included in the message body that is translated below, so perhaps they should be translated as well?

Reply:
@tfr-odoo What do you think? We should not send bad requests with faulty partner mails.

Reply:
@arj-odoo I think since we filter wrong email addresses before sending them to google, we indeed don't need to have a specific case for the error Invalid attendee email